src/controllers/order/index.ts

```diff
@@ -0,0 +1,54 @@
+import { Body, Tags, Security, ValidateError, SuccessResponse, Response, Controller, Post, Route } from 'tsoa'
+import { Logger } from 'pino'
+
+import { DemandRequest, DemandResponse } from '../../models'
+import { logger } from '../../lib/logger'
+import { BadRequest, NotFound } from '../../lib/error-handler'
+import Database from '../../lib/db'
+import { getMemberByAddress, getMemberBySelf } from '../../lib/services/identity'
+
+const SUBTYPE = 'order'
+const STATE = 'created'
+
+@Route('order')
+@Tags('order')
+@Security('bearerAuth')
+export class order extends Controller {
+  log: Logger
+  db: Database
+  constructor() {
+    super()
+    this.log = logger.child({ controller: '/order' })
+    this.db = new Database()
+  }
+
+  /**
+   * A Member creates a new demand for a order by referencing an uploaded parameters file.
+   * @summary Create a new order demand
+   * @param parametersAttachmentId The attachment's identifier
+   */
+  @Post()
+  @Response<BadRequest>(400, 'Request was invalid')
+  @Response<NotFound>(404, 'Demand was not found')
+  @Response<ValidateError>(422, 'Validation Failed')
+  @SuccessResponse('201')
+  public async createCapacity(@Body() { parametersAttachmentId }: DemandRequest): Promise<DemandResponse> {
+    const [attachment] = await this.db.getAttachment(parametersAttachmentId)
+    if (!attachment) throw new NotFound('attachment')
```

Comment:
seems like a `400` to me rather than a `404` as it's part of the request body not the URL

Reply:
but item is not found, there is another assertion in `order-demand.test.ts` file. @mattdean-digicatapult - thoughts?

It's clearly the attachment id not found and there is another assertion for ID not being UUID where we return validation error.

Reply:
Also it does not mean that it's a bad request, it's good, the item is not found so why we return bad request?

Reply:
Also we assert for request body not being UUID in the integration testing.

Reply:
maybe `422` then? `404` suggests the URL path is wrong, which it isn't (on the original request)

Reply:
WHen the URL path is wrong e.g. is not UUID there is a test where it will return ValidateError with 422 https://github.com/digicatapult/dscp-matchmaker-api/blob/056c6779b313533f3dbb3eac3f3e72d53b886814/test/integration/order-demand.test.ts#L48 you can find test scenario here

Reply:
why 404 suggest that path is wrong, it suggest that item is not found 

```
In computer network communications, the HTTP 404, 404 not found, 404, 404 error, page not found or file not found error message is a hypertext transfer protocol (HTTP) standard response code, to indicate that the browser was able to communicate with a given server, but the server could not find what was requested.
```
Quote ^^^ 
@mattdean-digicatapult - sorry but when you get a moment could please settle this down?

Reply:
Its 400.

Reply:
@dblane-digicatapult why it's 400? If we send a request with the UUID, knowing that it's in correct format, then what we do, we try to find an attachment, e.g. perform a db lookup for the given ID, the UUID/ID is in right format, so request itself is good? No?

Reply:
P.S still would like to hear why you wanted to use responses:
@dblane-digicatapult why 400 for item not found? (e.g. attachment) if the request URL contains UUID for the attachment that we do not have (server does not have)
@jonmattgray why 422 for the same reason?